### ISSUE TITLE
Trigger 404 route

### ISSUE THREAD

**Issue by sumanthratna** [bramus/router#111]
This is a great project, thanks!

Let's say I have the following:
```php
$router->mount('/blog', function() use ($smarty, $router, $posts) {

    // will result in '/blog/'
    $router->get('/', function() use ($smarty) {
        $smarty->display("pages/blog/index.tpl");
    });

    // will result in '/blog/id'
    $router->get('/([a-z0-9-]+)', function($id) use ($smarty, $posts) {
        if (isset($posts[$id])) {
            // successfully display correct post
        } else {
            // trigger 404 since the post doesn't exist
        }
    });

});
```
There's no guarantee that a post with ID `id` exists, which is why I have the `isset`. If there's no existing post, I'd like to be able to display a 404. Obviously, I could copy the code from `$router->set404`, or I could write the function for `$router->set404` separately and reference it in `$router->set404` and `$router->get('/([a-z0-9-]+)', function($id) ...`. However, I think code readability would be better with a `$router->trigger404()` (the method doesn't have to be named that, but you get the idea). 

Another thing I could do is loop through each of the possible `id` values and create a route as such, but IMO that also decreases code readability. 

TL;DR—trigger 404s inside of routes.

**Comment by sumanthratna** [comment]
https://github.com/bramus/router/issues/59#issuecomment-609995920 shows another way to address without modifying the library, but for code-readability and speed I think this should be implemented into bramus/router.

**Comment by bramus** [comment]
I'll gladly accept a PR that provides this.

**Comment by PlanetTheCloud** [comment]
Okay

**Comment by bramus** [comment]
This has been added into master by means of the `$router->trigger404()` method.

```php
$router->get('/([a-z0-9-]+)', function($id) use ($router) {
    if (!Posts::exists($id)) {
        $router->trigger404();
        return;
    }

    // …
});
```
